packages/openactive-integration-tests/test/helpers/flow-stages/b.js

```diff
@@ -52,6 +67,7 @@ async function runB({ templateRef, accessPass, brokerRole, uuid, sellerConfig, c
     brokerRole,
     positionOrderIntakeFormMap,
     customer,
+    paymentIdentifier: paymentIdentifierIfPaid,
```

Comment:
The other issue was, as you mentioned, Nick, that `paymentIdentifier` was being re-generated for each B/P call. Rather than using stage input/output, I've just put it as an arg to the B / P stages, which should make it very easy to do things like test what happens if the repeated B call has a different payment identifier, if a B-after-P call has a different payment identifier, etc